pilates/activitysim/preprocessor.py

```diff
@@ -72,21 +72,33 @@ def _create_skim_object(data_dir, overwrite=True):
     return True
 
 
-def _create_skims_by_mode(settings):
+def _create_skims_by_mode(settings, remote_url = None):
     """
     Returns 2 OD pandas dataframe for auto and transit
     """
     logger.info("Splitting BEAM skims by mode.")
-    skims_df = _load_raw_beam_skims(settings)
-
-    num_hours = skims_df['timePeriod'].nunique()
-    num_modes = skims_df['pathType'].nunique()
-    num_od_pairs = len(skims_df) / num_hours / num_modes
-
-    # make sure the matrix is square
-    num_taz = np.sqrt(num_od_pairs)
-    assert num_taz.is_integer()
-    num_taz = int(num_taz)
+    
+    
+    skims_df = _load_raw_beam_skims(settings, remote_url)
+    
+    ## JDC Comment: 
+    ## If OD pair has not skims, it is not included in the BeamSkims
+    ## we assume is zero (for transit). Although the assertion does not 
+    ## work, as it assumes a dataframe is fully populated. 
+
+#     num_hours = skims_df['timePeriod'].nunique()
+#     num_modes = skims_df['pathType'].nunique()
+#     num_od_pairs = len(skims_df) / num_hours / num_modes
```

Comment:
Isn't it worth checking that the skims are fully populated across time periods and mode?